### PR TITLE
Add PowerStatus tests and cleanup SystemInformation

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs [hunk 1]
```diff
@@ -71,51 +45,41 @@ public static bool DragFullWindows
...
         public static bool HighContrast
         {
             get
             {
                 EnsureSystemEvents();
                 if (systemEventsDirty)
                 {
-                    NativeMethods.HIGHCONTRAST_I data = new NativeMethods.HIGHCONTRAST_I();
-                    data.cbSize = Marshal.SizeOf(data);
+                    var data = new NativeMethods.HIGHCONTRAST_I();
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs]: Can you move this out of the cleanup commit? It's not really cleanup

File: src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs [hunk 2]
```diff
@@ -2057,19 +1430,15 @@ public static ScreenOrientation ScreenOrientation
             }
         }
 
-        /// <include file='doc\SystemInformation.uex' path='docs/doc[@for="SystemInformation.SizingBorderWidth"]/*' />
-        /// <devdoc>
-        ///    <para>
-        ///      Specifies the thikness, in pixels, of the Sizing Border.
-        ///    </para>
-        /// </devdoc>
+        /// <summary>
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs]: Would be even easier to review if you separated all of these <devdoc> to <summary> and removing the <include ...> to a different commit

**Comment by zsd4yr** [comment]
Good separation. Looking for a little more 😄 Otherwise LGTM

**Comment by hughbe** [comment]
Split into more commits. I'm also noting that changing a class with a private constructor to a static class is not considered a breaking change in .NET Core

File: src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs [hunk 3]
```diff
@@ -17,18 +17,18 @@ namespace System.Windows.Forms
...
     {
         // Figure out if all the multimon stuff is supported on the OS
-        private static bool checkMultiMonitorSupport = false;
-        private static bool multiMonitorSupport = false;
-        private static bool checkNativeMouseWheelSupport = false;
-        private static bool nativeMouseWheelSupport = true;
-        private static bool highContrast = false;
-        private static bool systemEventsAttached = false;
-        private static bool systemEventsDirty = true;
+        private static bool s_checkMultiMonitorSupport = false;
```
Comment (danmosemsft) [inline src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs]: The initialization to false is unnecessary, of course.

**Comment by danmosemsft** [comment]
LGTM. @zsd4yr are you implying you want to retain the commits (ie rebase or merge) or was that just for review? 

Beyond that I think tihs PR is good to go.

File: src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs [hunk 4]
```diff
@@ -1431,27 +1412,23 @@ public static Size MenuBarButtonSize
...
-
                 if (hDsk == IntPtr.Zero)
                 {
                     int error = Marshal.GetLastWin32Error();
-                    retVal = error == NativeMethods.ERROR_ACCESS_DENIED;
+                    return error == NativeMethods.ERROR_ACCESS_DENIED;
                 }
-
-                if (hDsk != IntPtr.Zero)
+                else
```
Comment (danmosemsft) [inline src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs]: there is no need for `else` since the `if` returns unconditionally.

**Comment by danmosemsft** [comment]
Ping @zsd4yr for question above.

This needs rebasing now anyway.

**Comment by hughbe** [comment]
Rebased... probably best reviewed commit-by-commit as-per (with ?w=1) at the end, but most changes are methodical

**Comment by zsd4yr** [comment]
> LGTM. @zsd4yr are you implying you want to retain the commits (ie rebase or merge) or was that just for review?
> 
> Beyond that I think tihs PR is good to go.

just for review

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PowerStatusTests.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+namespace System.Windows.Forms.Tests
+{
+    public class PowerStatusTests
+    {
+        [Fact]
+        public void PowerStatus_BatteryChargeStatus_Get_ReturnsExpected()
+        {
+            PowerStatus status = SystemInformation.PowerStatus;
+            Assert.True(Enum.IsDefined(typeof(BatteryChargeStatus), status.BatteryChargeStatus));
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PowerStatusTests.cs]: Just discovered, this test is failing is run on a laptop that is being charged:
image

**Pull request merged**